cpp/src/rolling/window_utils.cu

```diff
@@ -235,8 +235,10 @@ std::pair<std::unique_ptr<column>, std::unique_ptr<column>> windows_from_offset(
   rmm::device_async_resource_ref mr)
 {
   CUDF_FUNC_RANGE();
-  return detail::windows_from_offset(
-    input, length, offset, window_type, only_preceding, stream, mr);
+  auto result =
+    detail::windows_from_offset(input, length, offset, window_type, only_preceding, stream, mr);
+  stream.synchronize();
```

Comment:
I'm not seeing why this is needed. If it is necessary it should probably be in the detail function, right?

Reply:
I use a `rmm::exec_policy_nosync(stream)` execution policy for the `thrust::copy_n` call (of which there are two if `only_preceding` is `false`) to pipeline the calls in that case.

Reply:
That is correct. There are no copies or off-stream temporaries that I see so the caller can sync if they want.

Reply:
Removed.